dotnet/src/Connectors/Connectors.Memory.AzureCognitiveSearch/AzureCognitiveSearchMemoryStore.cs

```diff
@@ -428,5 +429,15 @@ private static T RunMemoryStoreOperation<T>(Func<T> operation)
         }
     }
 
+    private static double DistanceToSimilarity(double distance)
+    {
+        return 1 - (1 - distance) / distance;
```

Comment:
I'm not sure if this can be an issue, but if distance is 0, the division will explode...

Reply:
I've seen the same problem in all implementations of cosine similarity, when magnitude is 0 you get a division by zero.

Reply:
assuming correctness, why not return
`return 2 - 1 / distance` ?  Fewer operations.  Not sure how the C# compiler works, but possibly this could result in fewer operations and reads of distance variable?

Reply:
I renamed the function to "ScoreToSimilarity". When two vectors match, the score is 1, thus similarity 1

`1 - (1 - distance) / distance` => `1 - (1 - 1) / 1` => 1

Reply:
@momuno the operators precedence is different: `1 - (1 - distance) / distance` is `1 - ((1 - distance) / distance)` which could also be written as `(2 * distance -1) / distance`

Reply:
Not sure why I couldn't see this thread earlier.  Please see my note below about furthering the simplification of the expression.